viv_utils/idaloader.py

```diff
@@ -244,7 +245,7 @@ def loadWorkspaceFromIdb():
             raise RuntimeError('failed to read segment data')
 
         logger.debug('mapping section %s with %x bytes', segname, len(segbuf))
-        vw.addMemoryMap(segstart, envi.memory.MM_RWX, segname, segbuf)
+        vw.addMemoryMap(segstart, idautils.ida_segment.get_segm_by_name(segname).perm, filename, segbuf)
```

Comment:
the perm fix here looks good. can you explain using `segname` vs `filename`? wouldn't the former have more detail?

note: its been a while since i looked at this closely, so this may be a silly question.

Reply:
Vivisect searches information about the file (imports, exports, segments, etc.) by the filename.
If we use the segment's name we confuse vivisect because it doesn't know any file with the name of the segment.
The information about the file's segments is linked to the file information by its name in  the next line.
Also, the api requests the filename.

Reply:
ah, perfect, thanks!